### ISSUE TITLE
Testing set overfit

### ISSUE THREAD

**Issue by LouisEchard** [LouisEchard/ML_MiniProjects#20]
I am obviously overfitting my testing set. How come ?

Epoch:   0001 train cost= 1425.4730 test cost= 1563.5681
--
Epoch: 1001 train cost=   616.8939 test cost= 657.3666
Epoch: 2001 train cost=   473.9200 test cost= 471.5571
Epoch: 3001 train cost=   465.0009 test cost= 439.2402
Epoch: 4001 train cost=   423.5853 test cost= 418.4615
Epoch: 5001 train cost=   401.2415 test cost= 400.2988
Epoch: 6001 train cost=   407.7660 test cost= 387.8419
Epoch: 7001 train cost=   396.6616 test cost= 378.0742
Epoch: 8001 train cost=   365.2269 test cost= 370.9904
Epoch: 9001 train cost=   374.7301 test cost= 364.4301
Epoch: 10001 train cost=   380.3358 test cost= 357.2577
Epoch: 11001 train cost=   373.4594 test cost= 349.2540
Epoch: 12001 train cost=   359.1637 test cost= 342.3199
Epoch: 13001 train cost=   370.8046 test cost= 336.4300
Epoch: 14001 train cost=   351.7662 test cost= 331.2346
Epoch: 15001 train cost=   340.2113 test cost= 326.1385
Epoch: 16001 train cost=   342.3907 test cost= 319.5034
Epoch: 17001 train cost=   330.8310 test cost= 314.6164
Epoch: 18001 train cost=   328.2659 test cost= 306.9841
Epoch: 19001 train cost=   338.7257 test cost= 303.0684
Epoch: 20001 train cost=   309.0619 test cost= 298.6450
Epoch: 21001 train cost=   316.8529 test cost= 295.3780
Epoch: 22001 train cost=   312.7642 test cost= 291.9993
Epoch: 23001 train cost=   316.3476 test cost= 289.2612
Epoch: 24001 train cost=   318.3639 test cost= 285.9973
Epoch: 25001 train cost=   312.8478 test cost= 283.5542
Epoch: 26001 train cost=   289.6700 test cost= 280.5497
Epoch: 27001 train cost=   303.9113 test cost= 278.2597
Epoch: 28001 train cost=   297.8810 test cost= 275.3333
Epoch: 29001 train cost=   305.1180 test cost= 272.0818
Epoch: 30001 train cost=   302.1582 test cost= 270.1062
Epoch: 31001 train cost=   297.9511 test cost= 268.2026
Epoch: 32001 train cost=   294.2272 test cost= 264.7032
Epoch: 33001 train cost=   288.6759 test cost= 263.4919
Epoch: 34001 train cost=   284.8742 test cost= 261.3393
Epoch: 35001 train cost=   290.4778 test cost= 258.6602
Epoch: 36001 train cost=   279.5996 test cost= 256.9772
Epoch: 37001 train cost=   289.7794 test cost= 254.6733
Epoch: 38001 train cost=   275.1829 test cost= 252.5620
Epoch: 39001 train cost=   283.1600 test cost= 250.2829
Epoch: 40001 train cost=   284.9723 test cost= 249.7671
Epoch: 41001 train cost=   275.9490 test cost= 246.9826
Epoch: 42001 train cost=   276.0521 test cost= 244.6864
Epoch: 43001 train cost=   288.3893 test cost= 242.6662
Epoch: 44001 train cost=   265.4511 test cost= 240.7116
Epoch: 45001 train cost=   282.2848 test cost= 238.7856
Epoch: 46001 train cost=   259.1743 test cost= 235.9942
Epoch: 47001 train cost=   263.0318 test cost= 234.4595
Epoch: 48001 train cost=   271.1512 test cost= 233.0470
Epoch: 49001 train cost=   257.4264 test cost= 231.0835

By multiplying by 10 the neurons on both layers 👍 To :

Epoch: 0001 train cost= 1983.8573 test cost= 2012.2505
Epoch: 1001 train cost= 418.0538 test cost= 395.4067
Epoch: 2001 train cost= 371.9379 test cost= 359.7710
Epoch: 3001 train cost= 332.9927 test cost= 337.1420
Epoch: 4001 train cost= 313.4039 test cost= 310.5003
Epoch: 5001 train cost= 302.3638 test cost= 286.8333
Epoch: 6001 train cost= 298.4139 test cost= 274.6643
Epoch: 7001 train cost= 266.4141 test cost= 261.0793
Epoch: 8001 train cost= 264.2354 test cost= 251.5691
Epoch: 9001 train cost= 250.0905 test cost= 240.7923
Epoch: 10001 train cost= 245.2477 test cost= 233.6922
Epoch: 11001 train cost= 242.4936 test cost= 228.1841
Epoch: 12001 train cost= 243.7574 test cost= 220.9431
Epoch: 13001 train cost= 232.7052 test cost= 220.3625
Epoch: 14001 train cost= 218.5404 test cost= 212.0530
Epoch: 15001 train cost= 218.6976 test cost= 206.8828
Epoch: 16001 train cost= 235.9073 test cost= 202.0956
Epoch: 17001 train cost= 209.5925 test cost= 198.8741
Epoch: 18001 train cost= 219.4385 test cost= 196.4257
Epoch: 19001 train cost= 203.9956 test cost= 193.1953
Epoch: 20001 train cost= 210.8242 test cost= 190.7265
Epoch: 21001 train cost= 203.4394 test cost= 189.2151
Epoch: 22001 train cost= 196.6526 test cost= 186.2141
Epoch: 23001 train cost= 206.6519 test cost= 183.7675
Epoch: 24001 train cost= 187.1470 test cost= 181.5305
Epoch: 25001 train cost= 188.4347 test cost= 178.6632
Epoch: 26001 train cost= 194.1033 test cost= 177.2954
Epoch: 27001 train cost= 180.1917 test cost= 175.8273
Epoch: 28001 train cost= 182.3986 test cost= 173.3345
Epoch: 29001 train cost= 177.0954 test cost= 171.0396
Epoch: 30001 train cost= 171.7447 test cost= 168.6902
Epoch: 31001 train cost= 173.8187 test cost= 166.6871
Epoch: 32001 train cost= 172.7213 test cost= 164.1223
Epoch: 33001 train cost= 165.7559 test cost= 162.4485
Epoch: 34001 train cost= 156.3398 test cost= 159.3331
Epoch: 35001 train cost= 155.7690 test cost= 158.1360
Epoch: 36001 train cost= 163.8816 test cost= 155.1787
Epoch: 37001 train cost= 159.6441 test cost= 153.9702
Epoch: 38001 train cost= 149.5080 test cost= 151.1378
Epoch: 39001 train cost= 148.1086 test cost= 147.7042
Epoch: 40001 train cost= 143.7880 test cost= 148.5363
Epoch: 41001 train cost= 142.0380 test cost= 145.0098
Epoch: 42001 train cost= 145.3206 test cost= 144.0810
Epoch: 43001 train cost= 140.2040 test cost= 142.4639
Epoch: 44001 train cost= 134.1581 test cost= 139.9875

**Comment by LouisEchard** [comment]
Is my training set too similar to my testing set

**Comment by LouisEchard** [comment]
Do avoid this, while still working on NN architecture:
Get these variables (TTM, TimeOfDay) out. Check for overfit now. If now it's ok, then no worries. We first train our model without these variables and we will train later with transfer learning.

**Comment by LouisEchard** [comment]
Done now, no shuffle before train/test split and no TTM/TimeOfDay

**Comment by LouisEchard** [comment]
Epoch: 0001 train cost= 1557.9713 test cost= 2583.0481
Epoch: 1001 train cost= 1373.7395 test cost= 2361.0125
Epoch: 2001 train cost= 1275.1957 test cost= 2249.1948
Epoch: 3001 train cost= 1111.1788 test cost= 2127.6890
Epoch: 4001 train cost= 1063.9673 test cost= 2001.5109
Epoch: 5001 train cost= 939.7084 test cost= 1873.2107
Epoch: 6001 train cost= 783.3729 test cost= 1747.7007
Epoch: 7001 train cost= 710.5212 test cost= 1615.7842
Epoch: 8001 train cost= 616.2753 test cost= 1477.4216
Epoch: 9001 train cost= 517.5984 test cost= 1348.2202
Epoch: 10001 train cost= 454.4818 test cost= 1224.8394
Epoch: 11001 train cost= 374.9987 test cost= 1111.2085
Epoch: 12001 train cost= 348.9527 test cost= 1005.8768
Epoch: 13001 train cost= 316.6862 test cost= 908.4721
Epoch: 14001 train cost= 263.5651 test cost= 837.2226
Epoch: 15001 train cost= 256.3650 test cost= 789.9265
Epoch: 16001 train cost= 269.3628 test cost= 773.5729
Epoch: 17001 train cost= 261.0966 test cost= 769.9900
Epoch: 18001 train cost= 258.9889 test cost= 765.6155
Epoch: 19001 train cost= 249.9194 test cost= 770.3764
Epoch: 20001 train cost= 258.9730 test cost= 774.5306
Epoch: 21001 train cost= 251.1969 test cost= 772.6944
Epoch: 22001 train cost= 245.8822 test cost= 779.6662
Epoch: 23001 train cost= 253.5464 test cost= 780.3447
Epoch: 24001 train cost= 254.2510 test cost= 784.8536
Epoch: 25001 train cost= 253.0376 test cost= 794.0861
Epoch: 26001 train cost= 238.5863 test cost= 791.9638
Epoch: 27001 train cost= 242.4964 test cost= 793.6717
Epoch: 28001 train cost= 238.8524 test cost= 805.6860
Epoch: 29001 train cost= 238.3919 test cost= 810.2619
Epoch: 30001 train cost= 234.0022 test cost= 817.1363
Epoch: 31001 train cost= 243.8184 test cost= 820.0904
Epoch: 32001 train cost= 232.9232 test cost= 823.0692
Epoch: 33001 train cost= 218.3601 test cost= 828.5519
Epoch: 34001 train cost= 235.0771 test cost= 830.0101
Epoch: 35001 train cost= 243.6015 test cost= 836.9991
Epoch: 36001 train cost= 232.5037 test cost= 838.0209
Epoch: 37001 train cost= 225.5124 test cost= 844.3278
Epoch: 38001 train cost= 230.5441 test cost= 846.9470
Epoch: 39001 train cost= 233.6576 test cost= 847.4223
Epoch: 40001 train cost= 234.0341 test cost= 849.8903
Epoch: 41001 train cost= 233.8324 test cost= 854.7590
Epoch: 42001 train cost= 232.3525 test cost= 854.4861
Epoch: 43001 train cost= 222.8403 test cost= 859.1599
Epoch: 44001 train cost= 238.6243 test cost= 866.3644
Epoch: 45001 train cost= 237.9609 test cost= 864.9554
Epoch: 46001 train cost= 223.7048 test cost= 869.4374
Epoch: 47001 train cost= 220.2036 test cost= 877.2156
Epoch: 48001 train cost= 212.3552 test cost= 880.6426
Epoch: 49001 train cost= 217.1553 test cost= 880.4789
